stdlib/LinearAlgebra/src/qr.jl

```diff
@@ -542,6 +542,36 @@ function rank(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(
     return something(findfirst(i -> abs(A.R[i,i]) <= tol, 1:m), m+1) - 1
 end
 
+function nullspace(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(float(one(eltype(A.Q))))) * iszero(atol))
+    m, n = size(A, 1), size(A, 2)
+    (m == 0 || n == 0) && return Matrix{eigtype(eltype(A.Q))}(I, n, n)
+
+    indstart = rank(A; atol, rtol) + 1
+    return A.Q[:, indstart:end]
```

Comment:
If I recall correctly, we don't currently have efficient methods for slicing `Q` matrices (which are internally indirectly represented by linear operators).  See https://discourse.julialang.org/t/what-is-the-most-efficient-way-of-obtaining-the-orthogonal-column-space-of-a-matrix/72212/13

So you should probably need to do something like:
```jl
d = n + 1 - indstart # dimension of nullspace
I_slice = zeros(eltype(A), m, d) # construct last d columns of I
for i = 1:d; I_slice[i, indstart + d - 1] = 1; end
return A.Q * I_slice
```

Reply:
We do:

https://github.com/JuliaLang/julia/blob/98f3947a1d07d0a4644a69d161287a71246af37a/stdlib/LinearAlgebra/src/abstractq.jl#L90-L96